var/spack/repos/tutorial/packages/hdf5/package.py

```diff
@@ -104,7 +104,7 @@ class Hdf5(CMakePackage):
     if sys.platform != "darwin":
         depends_on("numactl", when="+mpi+fortran")
     depends_on("szip", when="+szip")
-    depends_on("zlib@1.1.2:")
+    depends_on("zlib-api")
```

Comment:
Like this:
```suggestion
    depends_on("zlib-api")
    depends_on("zlib@1.1.2:", when="^zlib")
```
I can go through and make them all as suggestions.

Reply:
In principle that would be better, but in this case I dropped it, since zlib 1.1.2 was probably released before I was born

Reply:
Yeah, that was just the first one I clicked on and I didn't even check the actual version :-D

How about I do this for all versions that are actually in spack, 1.2.3 and later? :-D